src/sql/parts/connection/connectionDialog/connectionDialogService.ts

```diff
@@ -313,6 +334,24 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		return this._dialogDeferredPromise;
 	}
 
+	public openDialogAndWaitButDontConnect(connectionManagementService: IConnectionManagementService,
+		params?: INewConnectionParams,
+		model?: IConnectionProfile,
+		connectionResult?: IConnectionResult): Thenable<IConnectionProfile> {
+
+		this.ignoreNextConnect = true;
```

Comment:
Small question - this seems to be only line different than openDialogAndWait (please corret me if I am misunderstanding). If so can we not just add a optional param in same function and update this value based on that?

Reply:
I could, I was trying to not change existing functionality since this code would probably have to be refactored anyways. But that would probably be a better solution.